[Settings] Shortcut edit dialog: Fix bugs

htcfreek opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

This PR fixes some bugs in our custom shortcut edit dialog:
   - Missing error badge in "invalid key" warning.
   - The dialog is not PT Settings theme aware.
   - Fixed a bug that when it is open but the window has no focus the key inputs are still catched.

Before:
![image](https://user-images.githubusercontent.com/61519853/228023416-865ae51b-2a3a-45ff-bd16-00239639516b.png)

After:
![image](https://user-images.githubusercontent.com/61519853/228023298-629a9063-1ed8-4e8b-b423-669f9f4bb371.png)



<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] **Closes:** #25059 , #25048, #25051
- [x] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [ ] **Tests:** Added/updated and all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

   - Added a resw comment for the shortcut instruction text. Currently the format markers gets lost on translation.

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

Local build and tested functionality.

niels9001 commented:
@htcfreek Would it make sense to replace this altogether with an InfoBar?

htcfreek commented:
> @htcfreek Would it make sense to replace this altogether with an InfoBar?

The badge/message yes. But I don't want to tuch to much of your code.

But then we have to split out the theme fix or you have to point me to the place where I need to add the infobar.

niels9001 commented:
> > @htcfreek Would it make sense to replace this altogether with an InfoBar?
> 
> @niels9001 
> The badge/message yes. But I don't want to tuch to much of your code.
> 
> But then we have to split out the theme fix or you have to point me to the place where I need to add the infobar.

Replacing the Border (and its content) with an InfoBar should do the trick (line61)!

https://github.com/microsoft/PowerToys/blob/21c959f0e3fcfb88899c5d510de06d45c1ae4eda/src/settings-ui/Settings.UI/Controls/ShortcutControl/ShortcutDialogContentControl.xaml#L61

Jay-o-Way commented:
> * The dialog is not theme aware.

Technically, it is, but it's using the App theme as in Windows Settings, where it should follow PowerToys Settings app theme.

htcfreek commented:
@niels9001
1. I have fixed the focus bug. (7877ef153834afc56cf889ebc88399a248abc0dd) - But when I add a break point to my event handling code it looks like something is playing ping-pong. I would expect that my code is only invoked once and not multiple times. Can you look at this or does it happen because of the break point?
2. And I replaced the custom info bar with the original info bar control. But the standard control is higher and therefore the UI gets bigger. I think your custom version fits much better.
![image](https://user-images.githubusercontent.com/61519853/228285664-907b75f4-7a7e-4bd6-9135-d8521d131122.png)

Jay-o-Way commented:
IMO the control is valid. Maybe we could move the secondary text _inside_ the infobar

htcfreek commented:
> IMO the control is valid. Maybe we could move the secondary text _inside_ the infobar

@Jay-o-Way 
Do you mean the text between infobar and buttons? This text is an always visible info text. So moving it inside the info bar makes no sense.

If @niels9001 is okay with the bigger you we can keep it.

Jay-o-Way commented:
It was just an idea. Actually, my thought is: It's a good hint, yet this message is **mostly** needed the moment the current shortcut is not valid. It could be either the .message, or the **.title** by itself. It does solve your point 3. :)

htcfreek commented:
> Also, even if the intention is the same, I think a sentence like "A shortcut must start with Windows, Ctrl, Alt or Shift key" would be just a bit better

Good point. I like this.

stefansjfw reviewed (approved):
LGTM! Nice catch for non-focused window!
